src/main/Main.hs

```diff
@@ -773,9 +788,9 @@ upgradeCmd (fromGit, repo) go = withConfigAndLock go $ globalFixCodePage go $
     upgrade (if fromGit then Just repo else Nothing) (globalResolver go)
 
 -- | Upload to Hackage
-uploadCmd :: ([String], Maybe PvpBounds) -> GlobalOpts -> IO ()
-uploadCmd ([], _) _ = error "To upload the current package, please run 'stack upload .'"
-uploadCmd (args, mpvpBounds) go = do
+uploadCmd :: ([String], Maybe PvpBounds, Bool) -> GlobalOpts -> IO ()
+uploadCmd ([], _, _) _ = error "To upload the current package, please run 'stack upload .'"
+uploadCmd (args, mpvpBounds, shouldSign) go = do
```

Comment:
I'd like to make it easier for people to automatically sign, likely with some setting in their stack.yaml file. Can that be added as a default in case `--sign` is not provided on the command line?